pkg/appmanager/resourceConfig_test.go

```diff
@@ -1005,133 +1005,408 @@ var _ = Describe("Resource Config Tests", func() {
+				},
+			}))
+			Expect(result.Actions).To(Equal([]*action{
+				&action{
+					Name:     "0",
+					HTTPHost: true,
+					Replace:  true,
+					Request:  true,
+					Value:    "newhost.com",
+				},
+				&action{
+					Name:    "1",
+					HTTPURI: true,
+					Path:    "/path",
+					Replace: true,
+					Request: true,
+					Value:   "/newpath",
+				},
+			}))
+
+			// route rewrite host
+			result = processURLRewrite("host.com", "newhost.com", routeType)
+			Expect(result.Conditions).To(Equal([]*condition{&condition{
+				Name:            "0",
+				CaseInsensitive: true,
+				Equals:          true,
+				External:        true,
+				Host:            true,
+				HTTPURI:         true,
+				Present:         true,
+				Remote:          true,
+				Request:         true,
+				Values:          []string{"host.com"},
+			}}))
+			Expect(result.Actions).To(Equal([]*action{&action{
+				Name:     "0",
+				HTTPHost: true,
+				Replace:  true,
+				Request:  true,
+				Value:    "newhost.com",
+			}}))
+
+			// route rewrite path
+			result = processURLRewrite("host.com/path", "/newpath", routeType)
+			Expect(result.Conditions).To(Equal([]*condition{
+				&condition{
+					Name:            "0",
+					CaseInsensitive: true,
+					Equals:          true,
+					External:        true,
+					Host:            true,
+					HTTPURI:         true,
+					Present:         true,
+					Remote:          true,
+					Request:         true,
+					Values:          []string{"host.com"},
+				},
+				&condition{
+					Name:        "1",
+					Equals:      true,
+					HTTPURI:     true,
+					Index:       1,
+					PathSegment: true,
+					Request:     true,
+					Values:      []string{"path"},
+				},
+			}))
+			Expect(result.Actions).To(Equal([]*action{&action{
+				Name:    "0",
+				HTTPURI: true,
+				Path:    "/path",
+				Replace: true,
+				Request: true,
+				Value:   "/newpath",
+			}}))
+
+			// route rewrite host and path
+			result = processURLRewrite("host.com/path", "newhost.com/newpath", routeType)
+			Expect(result.Conditions).To(Equal([]*condition{
+				&condition{
+					Name:            "0",
+					CaseInsensitive: true,
+					Equals:          true,
+					External:        true,
+					Host:            true,
+					HTTPURI:         true,
+					Present:         true,
+					Remote:          true,
+					Request:         true,
+					Values:          []string{"host.com"},
+				},
+				&condition{
+					Name:        "1",
+					Equals:      true,
+					HTTPURI:     true,
+					Index:       1,
+					PathSegment: true,
+					Request:     true,
+					Values:      []string{"path"},
+				},
+			}))
+			Expect(result.Actions).To(Equal([]*action{
+				&action{
+					Name:     "0",
+					HTTPHost: true,
+					Replace:  true,
+					Request:  true,
+					Value:    "newhost.com",
+				},
+				&action{
+					Name:    "1",
+					HTTPURI: true,
+					Path:    "/path",
+					Replace: true,
+					Request: true,
+					Value:   "/newpath",
+				},
+			}))
```

Comment:
You should do one test case that uses a multi-segmented path, to be sure that you get all the conditions for each segment.

Reply:
Added.